### PR TITLE
Rerun replaying mode after recording mode

### PR DESCRIPTION
**Pull request by zli82016**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/12366

For passed tests in recording mode, rerun 3 times in replaying mode to catch issues.

Built the new image `zhenhua-test-051901` and tested in PR https://github.com/GoogleCloudPlatform/magic-modules/pull/7950. Please check the comment in that PR.
https://github.com/GoogleCloudPlatform/magic-modules/pull/7950#issuecomment-1558679557

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by zli82016** [comment]
Screenshot 2023-05-23 at 7 07 53 PM

File: .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh [hunk 1]
```diff
@@ -217,13 +219,47 @@ if [[ -n $FAILED_TESTS_PATTERN ]]; then
...
+    gsutil -h "Content-Type:text/plain" -m -q cp testlog/replaying_build_after_recording/* gs://ci-vcr-logs/beta/refs/heads/auto-pr-$pr_number/artifacts/$build_id/build-log/replaying_build_after_recording/
+
+    # store replaying test logs
+    gsutil -h "Content-Type:text/plain" -m -q cp testlog/replaying_after_recording/* gs://ci-vcr-logs/beta/refs/heads/auto-pr-$pr_number/artifacts/$build_id/replaying_after_recording/
+
+    REPLAYING_FAILED_TESTS=$(grep "^--- FAIL: TestAcc" replaying_build_after_recording.log | sort -u -t' ' -k3,3 | awk -v pr_number=$pr_number -v build_id=$build_id '{print "`"$3"`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-"pr_number"/artifacts/"build_id"/build-log/replaying_build_after_recording/"$3"_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-"pr_number"/artifacts/"build_id"/replaying_after_recording/"$3".log)]"}')
+    if [[ -n $REPLAYING_FAILED_TESTS ]]; then
+      comment+="Tests failed when rerunning REPLAYING mode:${NEWLINE} $REPLAYING_FAILED_TESTS ${NEWLINE}${NEWLINE}"
+      comment+="Please fix these to complete your PR${NEWLINE}"
```
Comment (shuyama1) [inline .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh:254]: Do we want to explain what does failure in replaying mode mean? Or should we suggest they work with their reviewer to figure out the reason it failed in replaying after it passed in recording. Most of contributors may not know much about VCR tests

File: .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh [hunk 2]
```diff
@@ -217,13 +219,47 @@ if [[ -n $FAILED_TESTS_PATTERN ]]; then
...
+    for test in $RECORDING_PASSED_TEST_LIST
+    do
+      cat testlog/replaying_build_after_recording/${test}_replaying_test.log >> replaying_build_after_recording.log
+    done
+
+    # store replaying individual build logs
+    gsutil -h "Content-Type:text/plain" -m -q cp testlog/replaying_build_after_recording/* gs://ci-vcr-logs/beta/refs/heads/auto-pr-$pr_number/artifacts/$build_id/build-log/replaying_build_after_recording/
+
+    # store replaying test logs
+    gsutil -h "Content-Type:text/plain" -m -q cp testlog/replaying_after_recording/* gs://ci-vcr-logs/beta/refs/heads/auto-pr-$pr_number/artifacts/$build_id/replaying_after_recording/
```
[the remark this answers is not in the document]
Comment (zli82016) [reply .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh:249]: 1 debug log is expected for each test. Each test is run 3 times. The results for the 3 times are in one debug log.

File: .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh (hunk 1 above)
Comment (zli82016) [reply .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh:254]: I will add more information. Thanks for the comment.

Comment (zli82016) [reply .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh]: I added more information. Please check the current error message in the comment https://github.com/GoogleCloudPlatform/magic-modules/pull/8206#issuecomment-1610102995

File: .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh [hunk 3]
```diff
@@ -240,7 +277,7 @@ if [[ -n $FAILED_TESTS_PATTERN ]]; then
       # check for any uncaught errors in RECORDING mode
       comment+="Errors occurred during RECORDING mode. Please fix them to complete your PR${NEWLINE}"
     else
-      comment+="All tests passed${NEWLINE}"
+      comment+="All tests passed during RECORDING mode.${NEWLINE}"
```
Comment (shuyama1) [inline .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh:280]: Should we remove this? Will this comment be confusing to contributors if tests passed in recording but failed in replaying? Can you think of a case that we still need this line?

Comment (zli82016) [reply .ci/containers/gcb-terraform-vcr-tester/test_terraform_vcr.sh:280]: Good catch. The else block is not needed. I will remove these lines.

**Review by shuyama1 — APPROVED** [review]

**Comment by shuyama1** [comment]
Thanks for adding this feature!

**Pull request merged**